heareval/tasks/pipeline.py

```diff
@@ -548,12 +650,18 @@ def requires(self):
         }
 
     def run(self):
-        for audiofile in tqdm(list(self.requires()["corpus"].workdir.glob("*.wav"))):
-            # Compare the filename with the slug.
-            # Note that the slug does not have the extension of the file
-            split = self.metadata.loc[
-                self.metadata["slug"] == audiofile.stem, "split"
-            ].values[0]
+        audiofiles = list(self.requires()["corpus"].workdir.glob("*.wav"))
+        assert len(audiofiles) == len(
+            self.metadata["unique_filestem"].drop_duplicates()
+        )
```

Comment:
Just tested on nsynth and this assertion fails. I believe that metadata at this point still contains all the audio files prior to subsampling

Reply:
So maybe we want to also subsample the metadata when we are subsampling the splits. i.e. remove the audio files that we are not using from that split.

Reply:
No, just on small so far. Will test on large soon.

This happens in MonoWav... which is before SubsampleSplit. However this breaks on small datasets because not every metadata file has audio. I have relaxed the test

Reply:
Every metadata file in the small should have audio. We have this `metadata = metadata.loc[exists]` in the `ExtractMetadata` for small.